src/protocol/request_response/mod.rs

```diff
@@ -315,11 +326,6 @@ impl RequestResponseProtocol {
                 })
                 .await;
         }
-
-        // remove all pending inbound requests
-        for (request_id, _) in context.active_inbound {
-            self.pending_inbound_requests.remove(&(peer, request_id));
-        }
```

Comment:
So with the new logic we will handle the request anyway if the connection is closed, hitting an error when we try to send a response?

Reply:
Yep exactly, if the connection gets closed, we'll still have the substream in the FuturesUnordered and let them hit an error (either when we send out the request, or read the response) 🙏